### PR TITLE
[1LP][RFR] Fix some views elements

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
{{pytest: -k 'test_add_button or test_filter_save_and_load' --use-provider complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 87af77696910c40b69688cf9875f89d3376b9d3e:

`widgetastic_manageiq/__init__.py`:
- :bangbang: Line 1960:16: **E251** *unexpected spaces around keyword / parameter equals*
- :bangbang: Line 1960:18: **E251** *unexpected spaces around keyword / parameter equals*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2528e9dee61516490249d6f8bd1e96c4e8b8d5b8:

`cfme/services/catalogs/catalog_item.py`:
- :bangbang: Line 86:13: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/services/test_catalog_item.py`:
- :red_circle: Line 88:12: **F821** *undefined name 'appliance'*
- :red_circle: Line 89:8: **F821** *undefined name 'appliance'*
- :bangbang: Line 95:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b4edbfc9c861ea1cde198adab06b34803ddfac96:

`cfme/services/catalogs/catalog_item.py`:
- :heavy_exclamation_mark: Line 83:40: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 84:51: **W291** *trailing whitespace*
- :bangbang: Line 85:79: **E231** *missing whitespace after ','*
- :bangbang: Line 335:5: **E704** *multiple statements on one line (def)*
- :bangbang: Line 335:5: **E704** *multiple statements on one line (def)*
- :bangbang: Line 335:68: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 337:33: **E241** *multiple spaces after ','*
- :bangbang: Line 337:67: **E241** *multiple spaces after ','*
- :bangbang: Line 338:33: **E241** *multiple spaces after ','*
- :bangbang: Line 340:9: **E113** *unexpected indentation*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b34dccdbb592ece9145d15b30e5c8f94650d28da:

`cfme/services/catalogs/catalog_item.py`:
- :heavy_exclamation_mark: Line 83:40: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 84:51: **W291** *trailing whitespace*
- :bangbang: Line 85:79: **E231** *missing whitespace after ','*
- :heavy_exclamation_mark: Line 335:26: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 336:46: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 337:46: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 338:43: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 340:45: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 341:47: **W291** *trailing whitespace*
- :heavy_exclamation_mark: Line 343:45: **W291** *trailing whitespace*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/services/catalogs/catalog_item.py [hunk 1]
```diff
@@ -42,23 +44,77 @@ class RequestInfoTab(Tab, BasicProvisionFormView):
...
+
+
 class ButtonGroupForm(ServicesCatalogView):
     title = Text('#explorer_title_text')
 
     btn_group_text = Input(name='name')
     btn_group_hvr_text = Input(name='description')
-    btn_image = BootstrapSelect('button_image')
+    btn_image = VersionPick({Version.lowest(): BootstrapSelect('button_image'), "5.9": Text(
+        locator='//div[@id="button-icon-picker"]//span[@class="caret"]/ancestor::button')})
```
Comment (quarckster) [inline cfme/services/catalogs/catalog_item.py]: **REQUIRED**
For sake of readability
```python
btn_image = VersionPick({
    Version.lowest(): BootstrapSelect('button_image'),
    "5.9": Text(locator='//div[@id="button-icon-picker"]//span[@class="caret"]/ancestor::button')
})
```

File: cfme/services/catalogs/catalog_item.py [hunk 2]
```diff
@@ -42,23 +44,77 @@ class RequestInfoTab(Tab, BasicProvisionFormView):
     TAB_NAME = 'Request Info'
 
 
+class IconItem(ParametrizedView, ClickableMixin):
+    PARAMETERS = ('icon_title',)
+    ROOT = ParametrizedLocator('//div[@class = "tab-content"]//span/i[contains('
+                               '@title, {icon_title|quote})]')
+
+
+class ButtonImageSelectView(View):
```
Comment (quarckster) [inline cfme/services/catalogs/catalog_item.py]: **QUESTION**
Does that widget https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L3603 do the same?

**Comment by dajoRH** [comment]
Lint report for commit f01c7f22673be75a3b0366214b17e383c6dffa09:

`cfme/services/catalogs/catalog_item.py`:
- :red_circle: Line 6:1: **F401** *'widgetastic.widget.ClickableMixin' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/services/catalogs/catalog_item.py (hunk 2 above)
Comment (rlbabyuk) [reply cfme/services/catalogs/catalog_item.py]: Yes it is, @quarckster thanks a lot. I deleted unnecessary code

File: cfme/services/catalogs/catalog_item.py (hunk 1 above)
Comment (rlbabyuk) [reply cfme/services/catalogs/catalog_item.py]: done

File: cfme/services/catalogs/catalog_item.py [hunk 3]
```diff
@@ -265,28 +286,39 @@ def delete(self):
...
         view.fill({'btn_group_text': "group_text",
-                   'btn_group_hvr_text': "descr",
-                   'btn_image': "Button Image 1"})
+                   'btn_group_hvr_text': button_name,
+                   'btn_image': "Button Image 1" if self.appliance.version < '5.9' else 'broom'})
         view.add_button.click()
         view = self.create_view(DetailsCatalogItemView)
         assert view.is_displayed
-        view.flash.assert_success_message('Buttons Group "descr" was added')
+        return button_name
```
Comment (mshriver) [inline cfme/services/catalogs/catalog_item.py]: **Required**
I would like to keep a call to view.flash.assert_no_error() here - as you did below in add_button.

**Pull request merged**